src/main/java/greencity/service/impl/UserOwnSecurityServiceImpl.java

```diff
@@ -0,0 +1,58 @@
+package greencity.service.impl;
+
+import java.time.LocalDateTime;
+
+import greencity.dto.user_own_security.UserRegisterDto;
+import greencity.entity.User;
+import greencity.entity.UserOwnSecurity;
+import greencity.entity.enums.ROLE;
+import greencity.exception.BadEmailException;
+import greencity.repository.UserOwnSecurityRepo;
+import greencity.service.UserOwnSecurityService;
+import greencity.service.UserService;
+import greencity.service.VerifyEmailService;
+import lombok.AllArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
+public class UserOwnSecurityServiceImpl implements UserOwnSecurityService {
+
+    private UserOwnSecurityRepo repo;
+    private UserService userService;
+    private VerifyEmailService verifyEmailService;
+
```

Comment:
In my opinion DTO should be used at Controller layer. Service should receive pure Entity or data. 
But up tp U